cpp/src/arrow/compute/exec/aggregate_node.cc

```diff
@@ -165,7 +166,18 @@ class ScalarAggregateNode : public ExecNode {
   const char* kind_name() const override { return "ScalarAggregateNode"; }
 
   Status DoConsume(const ExecBatch& batch, size_t thread_index) {
+    util::tracing::Span span;
+    START_SPAN(span, "Consume",
+               {{"aggregate", ToStringExtra()},
+                {"node.label", label()},
+                {"batch.length", batch.length}});
     for (size_t i = 0; i < kernels_.size(); ++i) {
+      util::tracing::Span span;
+      START_SPAN(span, aggs_[i].function,
```

Comment:
Hmm, if we have the function name in the attribute, maybe the span name should be something generic like `ScalarAggregateConsume`?

Reply:
Or maybe not, we'll have to see what structure is best for analysis. (e.g. do we want to put node labels in the span names - maybe/probably, do we want function names in the span names - maybe not if it means we have too many distinct span names for visualization)